app/models/manageiq/providers/redfish/inventory/parser/physical_infra_manager.rb

```diff
@@ -20,6 +23,15 @@ def physical_servers
           :type            => "ManageIQ::Providers::Redfish::PhysicalInfraManager::PhysicalServer",
         )
         persister.computer_systems.build(:managed_entity => server)
+
+        enclosure_id = s.dig("Links", "Chassis", 0, "@odata.id")
+        next unless enclosure_id
+
+        rack = persister.physical_racks.lazy_find(enclosure_id)
+        server.assign_attributes(:physical_rack => rack)
```

Comment:
Hm since you're doing the `.build` right above you can probably just do this above and pass these in to the `.build` arg.